lib/shanty_docker_plugin/image_resolution_fallback.rb

```diff
@@ -0,0 +1,22 @@
+require 'shanty_docker_plugin/error'
+require 'shanty/logger'
+
+module DockerPlugin
+  # Methods for falling back to a different tag when resolving images
+  module ImageResolutionFallback
+    extend Shanty::Logger
+
+    module_function
+
+    def pull!(image, fallback_tag)
+      image.pull!
+    rescue Error::PullError
+      logger.warn(
```

Comment:
One thing I see that isn't captured in the move across to this plugin so far is the retries - that's something we'll have to think about how to cover.